ui-core/src/components/inputs/TolerancePicker.tsx

```diff
@@ -0,0 +1,129 @@
+/* eslint-disable react-hooks/exhaustive-deps */
+import React, { useState, useRef, useMemo, useEffect } from 'react';
+import Input, { InputProps } from './Input';
+import Modal from '../Modal';
+import cx from 'classnames';
+import { StatusWithMessage } from './StatusMessage';
+
+type ToleranceRange = { label: string; value: number };
+
+export type ToleranceValues = {
+  minusTolerance: number;
+  plusTolerance: number;
+};
+
+export type TolerancePickerProps = Omit<InputProps, 'value'> & {
+  onToleranceChange: (toleranceValues: ToleranceValues) => void;
+  toleranceRanges: ToleranceRange[];
+  toleranceValues?: ToleranceValues;
+};
+
+const TolerancePicker: React.FC<TolerancePickerProps> = ({
+  onToleranceChange,
+  toleranceRanges,
```

Comment:
I think it would be better to completely remove the toleranceRanges from the props (as we have discussed it, it will break the css etc).
```suggestion
```
You can move the toleranceRanges const from the story in the main file TolerancePicker. You can also rename it with screaming case TOLERANCE_RANGES